interface.py

```diff
@@ -0,0 +1,621 @@
+
+
+class Board(Frame):
+    def __init__(self, master, border_proportion=0.02):
+        self.main = master.main
+        Frame.__init__(self, master, bg=self.main.color_palette[0])
+        master.rowconfigure(0, weight=1)
+        master.columnconfigure(0, weight=1)
+        self.border_proportion = border_proportion
+        self._create_widgets()
+        self.bind("<Configure>", self._resize)
+
+    def _create_widgets(self):
+        self.content = []
+        self.board = []
+        self.border = Frame(self)
+        self.border.place(
+            relx=self.border_proportion,
+            rely=self.border_proportion,
+            relwidth=1 - 2 * self.border_proportion,
+            relheight=1 - 2 * self.border_proportion,
+        )
+        for x in range(8):
+            self.content.append([])
+            self.board.append([])
+            for y in range(8):
+                color = self.main.color_palette[9]
+                if (x + y) % 2 == 0:
+                    color = self.main.color_palette[2]
+                self.board[x].append("")
+                command = partial(self.replace, x, y)
+                self.content[x].append(
+                    Button(
+                        self.border,
+                        relief=FLAT,
+                        bg=color,
+                        activebackground=color,
+                        command=command,
+                        image=None,
+                    )
+                )
+                self.content[x][y].place(
+                    relx=0.125 * x, rely=0.125 * y, relwidth=0.125, relheight=0.125
+                )
+
+    def _resize(self, event):
+        for x in range(8):
+            for y in range(8):
+                if self.board[x][y] != "":
+                    image = self.getImage(self.board[x][y])
+                    self.content[x][y].config(image=image)
+                    self.content[x][y].image = image
+
+    def replace(self, x, y):
+        global selected_piece
+        global pieces_path
+        global home_path
+        if (self.board[x][y] == selected_piece or selected_piece == "") and self.board[
+            x
+        ][y] != "":
+            self.content[x][y].config(image="")
+            self.board[x][y] = ""
+        elif selected_piece != "":
+            self.board[x][y] = selected_piece
+            image = self.getImage(selected_piece)
+            self.content[x][y].config(image=image)
+            self.content[x][y].image = image
+        self.main.fen_placement = self.get_fen()
+        self.main.display_fen()
+
+    def get_size(self, num):
+        global pieces_sizes
+        global pieces_padding
+        if num <= pieces_sizes[0] + pieces_padding:
+            return pieces_sizes[0]
+        else:
+            for i in pieces_sizes[1:]:
+                if num < i + pieces_padding:
+                    return i
+        return pieces_sizes[-1]
+
+    def getImage(self, piece):

```

Comment:
getImage is not consistent with other names (that are snake_case)

Reply:
right